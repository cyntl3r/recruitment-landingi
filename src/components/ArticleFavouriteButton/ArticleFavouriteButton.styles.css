.button {
  background: #ff4747;
  padding: 7px 10px;
  color: #ffffff;
  font-size: 0.8rem;
  font-weight: bold;
  cursor: pointer;
  border: 0;
}
