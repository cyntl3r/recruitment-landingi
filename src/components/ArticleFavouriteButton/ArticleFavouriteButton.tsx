import { FunctionComponent } from 'react';
import styles from './ArticleFavouriteButton.styles.css';

export interface ArticleFavouriteButtonProps {
  isFavourite: boolean;
  onClick: () => void;
}

export const ArticleFavouriteButton: FunctionComponent<ArticleFavouriteButtonProps> = ({
  isFavourite,
  onClick,
}) => (
  <button
    data-testid="article-favourite-button"
    type="button"
    className={styles.button}
    onClick={onClick}
  >
    {isFavourite ? 'Remove from' : 'Add to'} {`"Favourites"`}
  </button>
);
