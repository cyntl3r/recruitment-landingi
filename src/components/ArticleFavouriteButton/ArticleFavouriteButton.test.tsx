import { render } from '../../testing/test-utils';
import { ArticleFavouriteButton } from './ArticleFavouriteButton';

test('snapshot', () => {
  const { container } = render(
    <ArticleFavouriteButton isFavourite={true} onClick={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});

test('favourite wrapper render', () => {
  const { getByTestId, findByDisplayValue } = render(
    <ArticleFavouriteButton isFavourite={true} onClick={jest.fn()} />
  );
  expect(getByTestId('article-favourite-button')).toBeDefined();
  expect(findByDisplayValue('Add to')).toBeTruthy();
});

test('no favourite wrapper render', () => {
  const { getByTestId, findByDisplayValue } = render(
    <ArticleFavouriteButton isFavourite={false} onClick={jest.fn()} />
  );
  expect(getByTestId('article-favourite-button')).toBeDefined();
  expect(findByDisplayValue('Remove from')).toBeTruthy();
});
