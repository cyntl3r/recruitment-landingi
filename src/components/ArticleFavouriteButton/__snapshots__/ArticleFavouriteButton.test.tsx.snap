// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <button
    data-testid="article-favourite-button"
    type="button"
  >
    Remove from
     
    "Favourites"
  </button>
</div>
`;
